pkg/client/client.go

```diff
@@ -932,6 +936,12 @@ func (c *Client) ValidatePrometheus(ctx context.Context, promNsName types.Namesp
 		return stop, nil
 	})
 
+	s, _ := c.kclient.AppsV1().StatefulSets("openshift-monitoring").Get(ctx, "prometheus-k8s", metav1.GetOptions{})
+	if c.lastGenerationNumber+10 < int(s.Status.ObservedGeneration) && s.Status.ReadyReplicas != s.Status.Replicas {
```

Comment:
I would leave a comment here explaining the reason behind the '+10' in the generation number. Probably also worth adding something in the Error message.

Reply:
You are right, I think now is clearer